backend/internal/activities/deployer_activities.go

```diff
@@ -690,6 +753,53 @@ func FetchKubeconfigStep(privateKeyPath string) ewf.StepFn {
 	}
 }
 
+func VerifyAddedNodeStep() ewf.StepFn {
+	return func(ctx context.Context, state ewf.State) error {
+		node, ok := state["node"].(kubedeployer.Node)
+		if !ok {
+			return fmt.Errorf("missing or invalid 'node' in state for verification")
+		}
+
+		kubeconfig, ok := state["kubeconfig"].(string)
```

Comment:
can we rebase after merging the cached pr and load the config from cache if found